TextFieldEffects/TextFieldEffects/HoshiTextField.swift

```diff
@@ -8,182 +8,252 @@
-        case .right:
-            originX += textRect.size.width - placeholderLabel.bounds.width
-        default:
-            break
-        }
-        placeholderLabel.frame = CGRect(x: originX, y: textRect.height/2,
-            width: placeholderLabel.bounds.width, height: placeholderLabel.bounds.height)
-        activePlaceholderPoint = CGPoint(x: placeholderLabel.frame.origin.x, y: placeholderLabel.frame.origin.y - placeholderLabel.frame.size.height - placeholderInsets.y)
-
-    }
-    
-    // MARK: - Overrides
-    
-    override open func editingRect(forBounds bounds: CGRect) -> CGRect {
-        return bounds.offsetBy(dx: textFieldInsets.x, dy: textFieldInsets.y)
-    }
-    
-    override open func textRect(forBounds bounds: CGRect) -> CGRect {
-        return bounds.offsetBy(dx: textFieldInsets.x, dy: textFieldInsets.y)
-    }
-    
+
+  /**
+   Fixed textfield height, change this only if designs change
+   */
+  @IBInspectable dynamic open var textFieldHeight: CGFloat = 45 {
+    didSet {
+      if let font = font, font.pointSize > CGFloat(15) {
+        height = textFieldHeight + 10
+      }
+    }
+  }
+
+  /**
+   The color of the border when it has no content.
+
+   This property applies a color to the lower edge of the control. The default value for this property is a clear color.
+   */
+  @IBInspectable dynamic open var borderInactiveColor: UIColor? {
+    didSet {
+      updateBorder()
+    }
+  }
+
+  /**
+   The color of the border when it has content.
+
+   This property applies a color to the lower edge of the control. The default value for this property is a clear color.
+   */
+  @IBInspectable dynamic open var borderActiveColor: UIColor? {
+    didSet {
+      updateBorder()
+    }
+  }
+
+  /**
+   The color of the placeholder text.
+
+   This property applies a color to the complete placeholder string. The default value for this property is a black color.
+   */
+  @IBInspectable dynamic open var placeholderColor: UIColor = .black {
+    didSet {
+      updatePlaceholder()
+    }
+  }
+
+  /**
+   The scale of the placeholder font.
+
+   This property determines the size of the placeholder label relative to the font size of the text field.
+   */
+  var placeholderFontSize: CGFloat = 12  {
+    didSet {
+      updatePlaceholder()
+    }
+  }
+  
+  override open var placeholder: String? {
+    didSet {
+      updatePlaceholder()
+    }
+  }
+
+  override open var bounds: CGRect {
+    didSet {
+      frame = CGRect(x: frame.origin.x,
+                     y: frame.origin.y,
+                     width: bounds.size.width,
+                     height: height)
+    }
+  }
+
+  override open func awakeFromNib() {
+    super.awakeFromNib()
+    initPlaceholderFont()
+  }
+
+  private let borderThickness: (active: CGFloat, inactive: CGFloat) = (active: 2, inactive: 0.5)
+  private let placeholderInsets = CGPoint(x: 0, y: 25)
+  private let textFieldInsets = CGPoint(x: 0, y: 6)
```

Comment:
A few of these properties are constants with initial values already set. That means they can't be configured later, is that correct?

Reply:
well those are fixed by design team, and there are two configurations for this text field, small and large, and the font and paddings are fixed.
I think I shouldn't make it configurable